src/ansys/sound/core/sound_composer/source_broadband_noise.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound Composer's broadband noise source."""
+import warnings
+
+from ansys.dpf.core import Field, FieldsContainer, Operator
+from matplotlib import pyplot as plt
+import numpy as np
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+from ._source_parent import SourceParent
+from .source_control_time import SourceControlTime
+
+ID_COMPUTE_LOAD_SOURCE_BBN = "sound_composer_load_source_bbn"
+ID_COMPUTE_GENERATE_SOUND_BBN = "sound_composer_generate_sound_bbn"
+
+
+class SourceBroadbandNoise(SourceParent):
+    """Sound Composer's broadband noise source class.
+
+    This class creates a broadband noise source for the Sound Composer. A broadband noise source is
+    used to generate a sound signal from a given broadband noise and its source control. The
+    broadband noise consists of a series of noise spectra corresponding each to a control parameter
```

Comment:
would suggest 'a series of noise spectra, each corresponding to a control parameter value.'

Reply:
done